csharp/rust/src/data.rs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+use crate::value::Value;
+use std::ffi::{
+    c_int, c_void,
+};
+use std::fmt::{Display, Formatter};
+use std::os::raw::c_char;
+use std::ptr::null;
+use std::str::Utf8Error;
+use tokio::runtime::Runtime;
+
+pub struct FFIHandle {
+    pub runtime: Runtime,
+    pub handle: crate::apihandle::Handle,
+}
+
+#[repr(C)]
+#[derive(PartialEq)]
+pub struct CreateClientHandleResult {
```

Comment:
As an alternative option you can use success/failure callback to pass status of client creation
Ref: #3411

Reply:
This PR removes the success/failure callbacks entirely in favor of embracing the dotnet task library in it's entirety.